pimd/pim_cmd_common.h

```diff
@@ -190,4 +192,47 @@ int pim_show_bsr_helper(const char *vrf, struct vty *vty, bool uj);
 	struct pim_instance *pimptr = vrfptr->info;                            \
 	MACRO_REQUIRE_SEMICOLON() /* end */
 
+/*
+ * Macros used to fix up the xpath so that global style PIM CLI handlers
+ * can continue to use the common functions which now expect "router pim"
+ * style configuration.
+ * Global PIM configuration was at the top-level config node, or
+ * within the vrf config node. Now configuration is moved to the
+ * router pim[6] [vrf NAME] node.
+ * So these macros will move vty to the router pim context and push
+ * the correct base xpath, then all the common config handling assumes
+ * that VTY_CURR_XPATH is correct and/or uses relative paths, and at
+ * the end vty is moved back to the original node and xpath popped
+ */
+#define START_PIM_DEPRECATED(clinode)                           \
```

Comment:
When and how are we going to get rid of this?

Reply:
Well, it could go away now, but I'd say once it's been long enough that no body remembers the previous PIM syntax then this stuff can be removed.